rpc/handlers.go

```diff
@@ -1038,3 +1038,48 @@ func (h *Handler) Call(call FunctionCall, id BlockID) ([]*felt.Felt, *jsonrpc.Er
 	}
 	return res, nil
 }
+
+func (h *Handler) EstimateFee(broadcastedTxns []BroadcastedTransaction, id BlockID) ([]*FeeEstimate, *jsonrpc.Error) {
+	var estimates []*FeeEstimate
+	state, closer, err := h.stateByBlockID(&id)
+	if err != nil {
+		return nil, ErrBlockNotFound
+	}
+	defer func() {
+		if closeErr := closer(); closeErr != nil {
+			h.log.Warnw("Failed to close state in starknet_call", "err", closeErr)
+		}
+	}()
+
+	header, err := h.blockHeaderByID(&id)
+	if err != nil {
+		return nil, ErrBlockNotFound
+	}
+
+	if header.GasPrice == nil {
```

Comment:
Should we just return `0` in this case?

Reply:
Nope, `header.GasPrice == nil` is a limitation of Juno. I added the `GasPrice` field to the block header without a migration, so some blocks don't have that field populated.

Reply:
Will it be possible to do a migration in the future that removes this limitation? It would be unfortunate to maintain this forever.

Reply:
We will remove it in the next release.